lume_model/variables.py

```diff
@@ -1,102 +1,128 @@
-    """
-    Base class used for constructing a scalar variable.
-
-    Attributes:
-        variable_type: Indicates scalar variable.
-        units: Units associated with scalar value.
-        parent_variable: Variable for which this is an attribute.
-    """
-    variable_type: str = "scalar"
-    units: Optional[str] = None  # required for some output displays
-    parent_variable: str = (
-        None  # indicates that this variable is an attribute of another
-    )
-
-
-class InputVariable(Variable, Generic[Value]):
-    """
-    Base class for input variables.
-
-    Attributes:
-        default: Default value assigned to the variable.
-        is_constant: Indicates whether the variable is constant.
-    """
-    default: Value  # required default
-    is_constant: bool = Field(False)
 
+    @property
+    @abstractmethod
+    def default_validation_config(self) -> dict[str, bool]:
+        """Determines default behavior during validation."""
+        return {}
 
-class OutputVariable(Variable, Generic[Value]):
-    """
-    Base class for output variables. Value and range assignment are optional.
+    @abstractmethod
+    def validate_value(self, value: Any, config: dict[str, bool] = None):
+        pass
 
-    Attributes:
-        default: Default value assigned to the variable.
-        value_range: Acceptable range for value.
-    """
-    default: Optional[Value] = None
-    value_range: Optional[list] = Field(None, alias="range")
+    def model_dump(self, **kwargs) -> dict[str, Any]:
+        config = super().model_dump(**kwargs)
+        return {"variable_class": self.__class__.__name__} | config
 
 
-class ScalarInputVariable(InputVariable[float], ScalarVariable):
-    """
-    Variable used for representing a scalar input. Scalar variables hold float values.
-    Initialization requires name, default, and value_range.
+class ScalarVariable(Variable):
+    """Variable for float values.
 
     Attributes:
-        value_range: Acceptable range for value.
-
-    Example:
-        ```
-        variable = ScalarInputVariable(name="example_input", default=0.1, value_range=[0.0, 1.0])
-        ```
-    """
-    value_range: list[float]
-
-
-class ScalarOutputVariable(OutputVariable[float], ScalarVariable):
+        default_value: Default value for the variable.
+        value_range: Value range that is considered valid for the variable. If the value range is set to None,
+          the variable is interpreted as a constant and values are validated against the default value.
+        value_range_tolerance: Absolute tolerance when checking whether values are within the valid range.
+        unit: Unit associated with the variable.
     """
-    Variable used for representing a scalar output. Scalar variables hold float values.
-    Initialization requires name.
-
-    Example:
-        ```
-        variable = ScalarOutputVariable(name="example_output")
-        ```
+    default_value: Optional[float] = None
```

Comment:
Was this intended to be optional? I think it will cause`torch_model._arrange_inputs` to fail when running:
> default_tensor = torch.tensor(
            [var.default_value for var in self.input_variables], **self._tkwargs
        )